### DISCUSSION TITLE
Recommendation on what to use?

### DISCUSSION THREAD

**Discussion by aar2416** [ConsenSys/quorum#1428] [Show and tell]
I want to create an application in which there should be an erc20 token that will represent an asset (not involving any currency). I want to build APIs for the registration of users and only those users will be able to access the erc20 token if it is minted by an admin for that user. 

For example, User logs in and purchases anything externally then the admin will know and the admin will use API to mint the erc20 token and transfer it to the user's account. Then the user can sell those erc20 tokens to any other user present in the chain. 

I want to know what consensus algorithm should I use? Does consensus algorithm mean I should have multiple admins to validate the transaction? How do I use hyperledger besu / GoQuorum to implement my private chain. There are lot of documentation available but I am not able to follow what steps to follow to achieve my solution. Will my solution be solvable using hyperledger besu/GoQuorum?

**Comment by baptiste-b-pegasys** [comment]
Hello, thank you for your question.

You may have a confusion on the consensus at the smart contract level (ERC20 or whatever), that is up to you (you can develop a voting, consensus algorithm on the smart contract), and the consensus at the protocol level, that is permissionned at the node level, that means only authorized nodes of the network will be able to participate on block generation.

Privacy on GoQuorum is also at the node level, with the public key of tessera, and it's not clear that you need privacy in your use case.

I think you can develop your application (dapp) with truffle and then deploy it later with ethereum (geth or besu) once you are ready.

https://forum.openzeppelin.com/t/create-an-erc20-using-truffle-without-writing-solidity/2713
https://www.toptal.com/ethereum-smart-contract/time-locked-wallet-truffle-tutorial

**Reply by aar2416** [reply]
Thanks for the reply. let me try this out.
